src/rules/relay_connection_arguments_spec.js

```diff
@@ -46,32 +46,70 @@ export function RelayConnectionArgumentsSpec(context) {
       }
 
       if (firstArgument) {
-        if (
-          firstArgument.type.kind != 'NamedType' ||
-          firstArgument.type.name.value != 'Int'
-        ) {
-          return context.reportError(
-            new ValidationError(
-              'relay-connection-arguments-spec',
-              'Fields that support forward pagination must include a `first` argument that takes a non-negative integer as per the Relay spec.',
-              [firstArgument]
-            )
-          );
+        if (hasBackwardPagination) {
+          // first and last must be nullable if both forward and backward pagination are supported.
+          if (
+            firstArgument.type.kind != 'NamedType' ||
+            firstArgument.type.name.value != 'Int'
+          ) {
+            return context.reportError(
+              new ValidationError(
+                'relay-connection-arguments-spec',
+                'Fields that support forward and backward pagination must include a `first` argument that takes a nullable non-negative integer as per the Relay spec.',
+                [firstArgument]
+              )
+            );
+          }
+        } else {
+          // first can be non-nullable if only forward pagination is supported.
+          const isFirstArgumentNullable =
+            firstArgument.type.kind !== 'NonNullType';
+          const type = isFirstArgumentNullable
+            ? firstArgument.type
+            : firstArgument.type.type;
+          if (type.kind != 'NamedType' || type.name.value != 'Int') {
```

Comment:
I think you can simplify this entire block by using the `unwrapType` function that is defined at the top of this file. 😄 

```js
const type = unwrapType(firstArgument);
```